src/buffers/disk/leveldb_buffer.rs

```diff
@@ -278,6 +295,9 @@ impl super::DiskBuffer for Buffer {
 
         let initial_size = db.value_iter(ReadOptions::new()).map(|v| v.len()).sum();
         let current_size = Arc::new(AtomicUsize::new(initial_size));
+        // We don't know what the real size is, but ,all things equal, it should have
+        // uniform distribution and one half of max is right in the middle of it.
+        let uncompacted_size = (initial_size as f64 * MAX_UNCOMPACTED * 0.5) as usize;
```

Comment:
We could also trigger compaction on startup so that we don't have to guess. It's likely redundant with the compaction we do when shutting down, but shouldn't hurt anything.

Reply:
True.